Automattic/jetpack
# ThreatsModal: Add bulk support

## Description
Update the original `ThreatModal` integration (proposed [here](https://github.com/Automattic/jetpack/pull/40202)) to add support for handling bulk threats.

**Note:** This addresses adding a modal for bulk fixer actions to align with that of the internal Protect alternative, however, we should consider a follow up task to add functionality to allow users so select any individual from the bulk list to view the individual threat details, proceed with actions, etc. Drafted up a potential solution for this here: https://github.com/Automattic/jetpack/pull/40720

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1561

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* See `ThreatsModal` and `ThreatsDataViews` storybooks
* Checkout this Protect branch on Jurassic Tube/Ninja
* Ensure that the modal is properly integrating in the `ScanResultsDataViews` instance and displays when trigger the `Show auto fixers` action

## Screenshots

Single threat:
![Screen Shot on 2024-12-19 at 19-18-13](https://github.com/user-attachments/assets/ceac96a1-e699-42c3-b52d-544fb6f83505)

Multiple threats:
![Screen Shot on 2024-12-20 at 14-08-05](https://github.com/user-attachments/assets/556ff1e0-642b-4b4d-93d9-607e5375ce9b)
